### ISSUE TITLE
get_interfaces_info with AIX

### ISSUE THREAD

**Issue by ulrichSchreiner** [ansible/ansible#12434]
hi,

i tried to use ansible 1.9.3 from a linux host against a AIX 6.1 host. the previous version worked, but now when gathering facts i get

```
 ...
  facts = ansible_facts(module)
  File "/var/tmp/ansible-tmp-1442561078.81-20680068205968/setup", line 4432, in ansible_facts facts.update(Network(module).populate())
  File "/var/tmp/ansible-tmp-1442561078.81-20680068205968/setup", line 3692, in populate interfaces, ips = self.get_interfaces_info(ifconfig_path)
TypeError: get_interfaces_info() takes exactly 3 arguments (2 given)
```

it looks like [this commit](https://github.com/ansible/ansible/commit/15ad02102ee79bf862d973041ae5f66dfb6884de) changed the signature of `get_interfaces_info` for the AIX platform (now there is a `ifconfig_options parameter`), but the superclass (`GenericBsdIfconfigNetwork`) calls this method with only one parameter. 

I'm not sure what the correct solution would be. Either set a default value for the parameter with `ifconfig_options='a'` or change the caller to give an additional parameter (whatever it would be for the `GenericBsdIfconfigNetwork`).

**Comment by Yannig** [comment]
The problem is also present in Ansible v2.

The following patch is doing the trick:

``` patch
diff --git a/lib/ansible/module_utils/facts.py b/lib/ansible/module_utils/facts.py
index ab37177..3d479c6 100644
--- a/lib/ansible/module_utils/facts.py
+++ b/lib/ansible/module_utils/facts.py
@@ -2355,7 +2355,7 @@ class AIXNetwork(GenericBsdIfconfigNetwork, Network):
         return interface['v4'], interface['v6']

     # AIX 'ifconfig -a' does not have three words in the interface line
-    def get_interfaces_info(self, ifconfig_path, ifconfig_options):
+    def get_interfaces_info(self, ifconfig_path, ifconfig_options='-a'):
         interfaces = {}
         current_if = {}
         ips = dict(
```

**Comment by ulrichSchreiner** [comment]
Would it be a problem to merge this fix to the latest 1.9? we still have 1.9.2 because this bug is still present in later 1.9-versions.

Upgrade to version2 is planned later this year.
